connman/plugins/sailfish_ofono.c

```diff
@@ -1197,6 +1224,10 @@ static void modem_delete(gpointer value)
 		g_source_remove(md->online_check_id);
 	}
 
+	if (md->pending_set_powered) {
+		g_cancellable_cancel(md->pending_set_powered);
```

Comment:
Cancellation happens asynchronously, via separate mainloop dispatch.  I suspect that that can lead to call to set_powered_cb(), which will be poking 'md' object already freed below. So, the callback might need to check that it is not dealing with G_IO_ERROR_CANCELLED error or stte before dereferencing md arg.

But it might just be me being confused with how all this cancellation stuff works in general / in context of libgofono, and this matches how other places in connman deal with similar cancellations -> I guess it is okay?

Reply:
Well in the core the `g_cancellable_cancel` is not used. Only in the plugins this is used so I guess it is ok to use here. But yeah, I was wondering that does this call the cb still for the cleanup? Apparently it can.